patches/server/0918-Add-damage-source-wrapper.patch

```diff
@@ -0,0 +1,1058 @@
++import io.papermc.paper.entity.damage.builder.DamageOriginBuilder;
++import net.kyori.adventure.text.Component;
++import net.minecraft.world.damagesource.DamageSource;
++import net.minecraft.world.damagesource.EntityDamageSource;
++import net.minecraft.world.damagesource.IndirectEntityDamageSource;
++import org.bukkit.craftbukkit.entity.*;
++import org.bukkit.entity.*;
++import org.jetbrains.annotations.Nullable;
++
++public class DamageOriginImpl implements DamageOrigin {
++    // TODO <!> keeps this list up to date with vanilla
++    // Vanilla
++    public static final DamageOrigin IN_FIRE = DamageOriginImpl.fromNMS(DamageSource.IN_FIRE);
++    public static final DamageOrigin LIGHTNING_BOLT = DamageOriginImpl.fromNMS(DamageSource.LIGHTNING_BOLT);
++    public static final DamageOrigin ON_FIRE = DamageOriginImpl.fromNMS(DamageSource.ON_FIRE);
++    public static final DamageOrigin LAVA = DamageOriginImpl.fromNMS(DamageSource.LAVA);
++    public static final DamageOrigin HOT_FLOOR = DamageOriginImpl.fromNMS(DamageSource.HOT_FLOOR);
++    public static final DamageOrigin IN_WALL = DamageOriginImpl.fromNMS(DamageSource.IN_WALL);
++    public static final DamageOrigin CRAMMING = DamageOriginImpl.fromNMS(DamageSource.CRAMMING);
++    public static final DamageOrigin DROWN = DamageOriginImpl.fromNMS(DamageSource.DROWN);
++    public static final DamageOrigin STARVE = DamageOriginImpl.fromNMS(DamageSource.STARVE);
++    public static final DamageOrigin CACTUS = DamageOriginImpl.fromNMS(DamageSource.CACTUS);
++    public static final DamageOrigin FALL = DamageOriginImpl.fromNMS(DamageSource.FALL);
++    public static final DamageOrigin FLY_INTO_WALL = DamageOriginImpl.fromNMS(DamageSource.FLY_INTO_WALL);
++    public static final DamageOrigin OUT_OF_WORLD = DamageOriginImpl.fromNMS(DamageSource.OUT_OF_WORLD);
++    public static final DamageOrigin GENERIC = DamageOriginImpl.fromNMS(DamageSource.GENERIC);
++    public static final DamageOrigin MAGIC = DamageOriginImpl.fromNMS(DamageSource.MAGIC);
++    public static final DamageOrigin WITHER = DamageOriginImpl.fromNMS(DamageSource.WITHER);
++    public static final DamageOrigin ANVIL = DamageOriginImpl.fromNMS(DamageSource.ANVIL);
++    public static final DamageOrigin FALLING_BLOCK = DamageOriginImpl.fromNMS(DamageSource.FALLING_BLOCK);
++    public static final DamageOrigin DRAGON_BREATH = DamageOriginImpl.fromNMS(DamageSource.DRAGON_BREATH);
++    public static final DamageOrigin DRY_OUT = DamageOriginImpl.fromNMS(DamageSource.DRY_OUT);
++    public static final DamageOrigin SWEET_BERRY_BUSH = DamageOriginImpl.fromNMS(DamageSource.SWEET_BERRY_BUSH);
++    public static final DamageOrigin FREEZE = DamageOriginImpl.fromNMS(DamageSource.FREEZE);
++    public static final DamageOrigin FALLING_STALACTITE = DamageOriginImpl.fromNMS(DamageSource.FALLING_STALACTITE);
++    public static final DamageOrigin STALAGMITE = DamageOriginImpl.fromNMS(DamageSource.STALAGMITE);
++
```

Comment:
These should be exposed somewhere else, as this isn't reachable via the api.

Reply:
I agree but i need to convert the nms into paper api, and i'm not a big fan of using reflection like you did in your impl adapter